clients/fides-js/src/lib/consent.tsx

```diff
@@ -0,0 +1,173 @@
+  if (options === undefined || typeof options !== "object") {
+    return false;
+  }
+
+  if (typeof options.labels === "object") {
+    let validLabels = true;
+    Object.entries(options.labels).forEach((value: [string, string]) => {
+      if (typeof value[1] !== "string") {
+        debugLog(`Invalid banner options: labels.${value[0]} is not a string!`);
+        validLabels = false;
+      }
+    });
+
+    if (!validLabels) {
+      return false;
+    }
+  }
+
+  if (!options.privacyCenterUrl) {
+    debugLog("Invalid banner options: privacyCenterUrl is required!");
+    return false;
+  }
+
+  if (options.privacyCenterUrl) {
+    try {
+      // eslint-disable-next-line no-new
+      new URL(options.privacyCenterUrl);
+    } catch (e) {
+      debugLog(
+        "Invalid banner options: privacyCenterUrl is an invalid URL!",
+        options
+      );
+      return false;
+    }
+  }
+
+  return true;
+};
+
+/**
+ * Fetch the user's geolocation from an external API
+ */
+const getLocation = async (): Promise<UserGeolocation> => {
+  // assumes that isGeolocationEnabled is true
+  debugLog("Running getLocation...");
+  const options = getBannerOptions();
+  const {geolocationApiUrl} = options
+
+  if (!geolocationApiUrl) {
+    debugLog(
+        "Location cannot be found due to no configured geoLocationApiUrl."
+    );
+    return {};
+  }
+
+  debugLog(`Calling geolocation API: GET ${geolocationApiUrl}...`);
+  const fetchOptions: RequestInit = {
+    mode: "cors",
+  };
+  const response = await fetch(geolocationApiUrl, fetchOptions);
+
+  if (!response.ok) {
+    debugLog(
+      "Error getting location from geolocation API, returning {}. Response:",
+      response
+    );
+    return {};
+  }
+
+  try {
+    const body = await response.json();
+    debugLog("Got location response from geolocation API, returning:", body);
+    return body;
+  } catch (e) {
+    debugLog(
+      "Error parsing response body from geolocation API, returning {}. Response:",
+      response
+    );
+    return {};
+  }
+};
+
+/**
+ * Initialize the Fides Consent Banner or Link, with optional extraOptions to override defaults.
+ *
+ * (see the type definition of ConsentBannerOptions for what options are available)
+ */
+export const initFidesConsent = async (
+  defaults: CookieKeyConsent,
+  extraOptions?: ConsentBannerOptions
+): Promise<void> => {
+  debugLog(
+    "Initializing Fides consent banner with consent cookie defaults...",
+    defaults
+  );
+  if (extraOptions !== undefined) {
+    if (typeof extraOptions !== "object") {
+      // eslint-disable-next-line no-console
+      console.error(
+        "Invalid 'extraOptions' arg for Fides.banner(), ignoring",
+        extraOptions
+      );
+    } else {
+      // If the user provides any extra options, override the defaults
+      setBannerOptions({ ...getBannerOptions(), ...extraOptions });
+    }
+  }
+  const options: ConsentBannerOptions = getBannerOptions();
```

Comment:
TBH this is pretty sloppy initialization logic. I recognize it as my own so I can say that 😄 

This version of the banner is looking _way_ more maintainable with Preact in place, so we can definitely remove some of the unnecessary footguns here with state management. You'd be able to delete all of this logic if the `ConsentBanner` component itself would just manage merging options with defaults - that's standard React stuff that won't require this back-breaking API 👍

Reply:
See https://github.com/ethyca/fides/pull/3191#discussion_r1188869210. My main question is how to manage configs for more global vars like those related to geolocation (happens outside the banner component)

Reply:
Replied over there, but to double-down: I think we need a `Fides.options` object we can put all these things in 👍